scripts/west_commands/ncs_commands.py

```diff
@@ -17,17 +17,56 @@
 
 import ncs_west_helpers as nwh
 
+def add_zephyr_rev_arg(parser):
+    parser.add_argument('-z', '--zephyr-rev',
+                        help='''upstream git ref to use for zephyr project;
+                        default is upstream/master''')
+
+def add_projects_arg(parser):
+    parser.add_argument('projects', metavar='PROJECT', nargs='*',
+                        help='''projects (by name or path) to operate on;
+                        defaults to all projects in the manifest which
+                        are shared with the upstream manifest''')
+
+def check_west_version():
+    min_ver = '0.6.99'
+    if (packaging.version.parse(west_ver) <
+            packaging.version.parse(min_ver)):
+        log.die('this command currently requires a pre-release west '
+                f'(your west version is {west_ver}, '
+                f'but must be at least {min_ver}')
+
+def likely_merged(np, zp, nsha, zsha):
+    analyzer = nwh.RepoAnalyzer(np, zp, nsha, zsha)
+    likely_merged = analyzer.likely_merged
+    if likely_merged:
+        # likely_merged is a map from downstream commits to
+        # lists of upstream commits that look similar.
+        log.msg('downstream patches which are likely merged upstream',
+                '(revert these if appropriate):', color=log.WRN_COLOR)
+        for dc, ucs in likely_merged.items():
+            log.inf(f'- {dc.oid} ({nwh.commit_shortlog(dc)})\n'
+                    '  Similar upstream commits:')
+            for uc in ucs:
+                log.inf(f'  {uc.oid} ({nwh.commit_shortlog(uc)})')
+    else:
+        log.inf('no downstream patches seem to have been merged upstream')
+
+def to_ncs_name(zp):
+    # convert zp, a west.manifest.Project in the zephyr manifest,
+    # to the equivalent Project name in the NCS manifest.
+    #
+    # TODO: does the fact that this is needed mean the west commit
+    #       which names the manifest project 'manifest' unconditionally
+    #       was wrong to do so?
+    if zp.name == 'manifest':
```

Comment:
if all imported manifest project are named 'manifest', we could end up into some mess as all those are name `manifest`, although they are in different imported projects.
Is the reason for manifest name, that the toplevel project may not have `self::path` defined, and thus we cannot know the name (unless we simply use the folder name, if `self::path` is not set)
For imported projects, we could use the name specified in the manifest for the project being imported.

Should manifest be an attribute on a project, to indicate that it is an imported project and holds a `manifest` file itself ?
Each manifest project has a

Reply:
I don't follow here, we don't have a project named `manifest` in our manifest. Why is `zp.name` possibly set to `manifest`?

Reply:
Well, as you know, I want to get rid of ManifestProject entirely; see https://github.com/zephyrproject-rtos/west/issues/327.

This is yet more evidence that having the manifest itself as an object inside manifest.projects is a confusing mistake in my opinion :).

But for now, we do, and it's always named "manifest", which is a reserved name that no project can possibly have. The manifest.py parsing code checks that and rejects any attempt to name a project 'manifest'.

Either way I think this is more of a question about west than this particular PR, right?

Reply:
Let's discuss this in west itself, since as @mbolivar describes, there are already plans to get rid of this project in memory as part of the project list.

Reply:
@carlescufi completely agree.
Was not aware that the project name `manifest` was actually a side effect / would be fixed by https://github.com/zephyrproject-rtos/west/issues/327.